### PR TITLE
Bookstore: Use Redux in React components.

### PR DESCRIPTION
**Pull request by SebastianSolisFenger**
## PR changes:

- Connect react with redux using its official binding react-redux
- Provide the store to the application tree using the Provider component of react-redux.
- Implement register book functionality by dispatching the "add book" action in the store.
- Implement delete book functionality by dispatching the "remove book" action in the store.
- Implement check status in the categories page by dispatching the "check status" action in the store.

### CODE REVIEW

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @SebastianSolisFenger , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
Linter is green (the last commit have a green check mark (✅)) 👏
README file is descriptive 👏
Commit messages are meaningful 👏 👏
Github flow is correctly used (Pull Request is made from the feature branch into the main branch) 👏

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**You can also consider**:

- [ ] Import `applyMiddleware` - to use some handy Redux middleware, like a redux logger.

- it will help a lot in the development process. To install it, you needs to run:
  `npm i --save redux-logger`
  

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Categories.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  };
+
+  return (
+    <>
+      <button type="button" onClick={checkStatusEventHandler}>
+        Check status
+      </button>
+    </>
+  );
+};
```
[the remark this answers is not in the document]
Comment (SebastianSolisFenger) [reply src/components/Categories.js:19]: Hi @OLIPLICHE. I appreciate your comment but I don't understand what you mean. Isn't it already despatching it?
 the action from src/redux/categories/categories.js

[the remark this answers is not in the document]
Comment (SebastianSolisFenger) [reply src/components/Categories.js:19]: @OLIPLICHE .. As you can see, it's on  src/redux/categories/categories.js. 

// ACTION TYPES FOR CATEGORIES
const CHECK_STATUS = 'bookStore/books/CHECK_STATUS';

// ACTION CREATORS
export const checkStatus = () => ({ type: CHECK_STATUS });

const categories = (state = [], action) => {
  switch (action.type) {
    case CHECK_STATUS:
      return 'Under Construction';
    default:
      return state;
  }
};

export default categories;

[the remark this answers is not in the document]
Comment (OLIPLICHE) [reply src/components/Categories.js:19]: - [ ] The App does not display `Under Construction` when I click `Check status` button. Kindly make this appears on screen

**Pull request merged**